diff --git a/src/core/render/renderframe.cpp b/src/core/render/renderframe.cpp
index 6fb2799..89f1318 100644
--- a/src/core/render/renderframe.cpp
+++ b/src/core/render/renderframe.cpp
@@ -107,17 +107,6 @@ const std::vector<vk::UniqueDescriptorSet>& RenderFrame::getDescriptorSets() con
 	return descriptorSets;
 }
 
-void RenderFrame::addUniformBufferBlock(BufferBlock* block)
-{
-    uniformBufferBlocks.push_back(block);
-}
-
-BufferBlock* RenderFrame::getUniformBufferBlock(int index) 
-{
-    assert(index < uniformBufferBlocks.size());
-    return uniformBufferBlocks[index];
-}
-
 }; // namespace render
 
 }; // namespace sword
diff --git a/src/core/render/renderframe.hpp b/src/core/render/renderframe.hpp
index eb3cbc2..13e8fbc 100644
--- a/src/core/render/renderframe.hpp
+++ b/src/core/render/renderframe.hpp
@@ -43,8 +43,7 @@ public:
     void addRenderLayer(RenderLayer&&);
     void addRenderLayer(const RenderPass&, const GraphicsPipeline&, const vk::Device& device);
     void clearRenderPassInstances();
-    void addUniformBufferBlock(BufferBlock*);
-    BufferBlock* getUniformBufferBlock(int index);
+    BufferBlock* bufferBlock;
 
 private:
     std::unique_ptr<Attachment> swapchainAttachment;
@@ -54,7 +53,6 @@ private:
     vk::UniqueFence fence;
     vk::UniqueSemaphore semaphore;
     std::vector<vk::UniqueDescriptorSet> descriptorSets;
-    std::vector<BufferBlock*> uniformBufferBlocks;
     void updateDescriptorSet(uint32_t setId, const std::vector<vk::WriteDescriptorSet>);
     uint32_t width;
     uint32_t height;
